mobile/lib/modules/search/ui/immich_search_bar.dart

```diff
@@ -21,6 +22,28 @@ class ImmichSearchBar extends HookConsumerWidget
     final searchTermController = useTextEditingController(text: "");
     final isSearchEnabled = ref.watch(searchPageStateProvider).isSearchEnabled;
 
+    focusSearch() {
+      searchTermController.clear();
+      ref.watch(searchPageStateProvider.notifier).getSuggestedSearchTerms();
+      ref.watch(searchPageStateProvider.notifier).enableSearch();
+      ref.watch(searchPageStateProvider.notifier).setSearchTerm("");
+
+      searchFocusNode.requestFocus();
+    }
+    onSearchFocusRequest() {
+      focusSearch();
+    }
+
+    useEffect(
+      () {
+        searchFocusNotifierProvider.addListener(onSearchFocusRequest);
```

Comment:
I guess we can use the `focusSearch` method directly as the callback to the notifier?

Reply:
Correct. I think its cleaner. I was having a hard time getting used to the codebase and used this naming to keep things readable, intent-wise :)